rosidl_generator_c/resource/msg__functions.c.em

```diff
@@ -235,14 +237,15 @@ for line in lines:
 @(message_typename) *
 @(message_typename)__create()
 {
-  @(message_typename) * msg = (@(message_typename) *)malloc(sizeof(@(message_typename)));
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
```

Comment:
> in some scenarios is not feasible to use the common dynamic memory API (malloc, free,...).

I think the default allocator uses `malloc()` and `free()`, and this appears to give no way to change which allocator is used. The `__create()` function would need to accept an allocator as an argument for that to be changed.

I would also recommend adding an `rcutils_allocator_t` to the message struct so the `__destroy()` function is guaranteed to use the same allocator.

>  is there any possibility of backporting this to Foxy?

If we modify the function signature then this can't be backported because it breaks ABI. If hard coding the default allocator is good enough for your use case (maybe it's enough to be able to `LD_PRELOAD` a replacement for `rcutils_get_default_allocator()`?) then it seems like that could be backported.

Reply:
Regarding the allocation functions that are retrieved with `rcutils_get_default_allocator` that's true that modifying the default allocators can be tricky.
But by now, in micro-ROS, we use a slightly modified version of the `rcutils` (stunned for embedded purposes) that has [`rcutils_set_default_allocator`](https://github.com/micro-ROS/rcutils/blob/1fdfd3cd2b645ce683fe3e150feeac7c4a1bf073/src/allocator.c#L87) functionality. We re planning to PR some of these changes to the `rcutils` mainline.

So, in that sense, in our opinion its ok to retrieve a global-wise allocator with `rcutils_get_default_allocator` and ABI/API is not modified so we can backport.

Reply:
Ah, that makes sense. Mind opening an issue to change these APIs to accept a non-default allocator in the future?

Reply:
I'm interested in being able to pass a non-default allocator. I think we can use this existing issue to track progress: https://github.com/ros2/rosidl/issues/306